### PR TITLE
feat: rebrand Runnable to AppSurface

### PR DESCRIPTION
**Pull request by akrock**
## Summary

- Rebrands the project and package surface from Runnable to AppSurface, with RazorWire preserved as the reactive Razor UI package family.
- Renames solution, projects, namespaces, package ids, docs, workflows, and generated package chooser content for the new public packaging story.
- Adds package publishing metadata and package gate validation so public, support, proof-host, and excluded packages are explicit before NuGet publishing.
- Keeps the RazorWire CLI excluded from publishing until its .NET tool packaging is ready, and hardens the new package gate workflow.

## Validation

- `dotnet test tools/ForgeTrust.AppSurface.PackageIndex.Tests/ForgeTrust.AppSurface.PackageIndex.Tests.csproj`
- `dotnet run --project tools/ForgeTrust.AppSurface.PackageIndex/ForgeTrust.AppSurface.PackageIndex.csproj -- gate`
- `dotnet build ForgeTrust.AppSurface.slnx --no-restore`
- `dotnet pack ForgeTrust.AppSurface.slnx --no-build --output /private/tmp/appsurface-pack-smoke`
- `dotnet run --project tools/ForgeTrust.AppSurface.PackageIndex/ForgeTrust.AppSurface.PackageIndex.csproj -- verify-packages --package-version 0.1.0-ci.local --artifacts-output /private/tmp/appsurface-package-artifacts --report /private/tmp/appsurface-package-artifacts/package-validation-report.md`
- `dotnet format ForgeTrust.AppSurface.slnx --verify-no-changes --no-restore`
- stale Runnable/AppSurface naming scans with `rg`
- `git diff --check`

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
     about: Read how to file focused feedback, propose docs changes, and prepare pull requests.
   - name: Start with the package chooser
-    url: https://github.com/forge-trust/Runnable/blob/main/packages/README.md
-    about: Use this when you are deciding which Runnable package to install first.
+    url: https://github.com/forge-trust/AppSurface/blob/main/packages/README.md
+    about: Use this when you are deciding which AppSurface package to install first.
   - name: Read the release and upgrade contract
-    url: https://github.com/forge-trust/Runnable/blob/main/releases/README.md
+    url: https://github.com/forge-trust/AppSurface/blob/main/releases/README.md
     about: Use this when you are evaluating adoption risk or migration guidance.
```
[the remark this answers is not in the document]
Comment (akrock) [reply .github/ISSUE_TEMPLATE/config.yml:14]: Intentionally keeping these links on `forge-trust/AppSurface`. The repository has moved/been renamed to AppSurface, and the PR is now under https://github.com/forge-trust/AppSurface/pull/268. We do not want to revert these URLs back to `forge-trust/Runnable`.

File: SECURITY.md [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 
-Runnable is pre-`v1.0.0`, but security reports still need a private path.
+AppSurface is pre-`v1.0.0`, but security reports still need a private path.
 
 ## Reporting a vulnerability
 
 Do not open a public GitHub issue for suspected vulnerabilities, leaked secrets, exploit details, or reports that include sensitive deployment information.
 
-Use GitHub's private security advisory flow instead: [report a vulnerability privately](https://github.com/forge-trust/Runnable/security/advisories/new).
+Use GitHub's private security advisory flow instead: [report a vulnerability privately](https://github.com/forge-trust/AppSurface/security/advisories/new).
```
[the remark this answers is not in the document]
Comment (akrock) [reply SECURITY.md:9]: Intentionally keeping this private advisory link on `forge-trust/AppSurface`. The repository is being updated to AppSurface, and GitHub is already serving this PR from https://github.com/forge-trust/AppSurface/pull/268. Reverting this to `forge-trust/Runnable` would be the wrong direction for the rebrand.

File: ForgeTrust.AppSurface.Core.Tests/DefaultEnvironmentProviderTests.cs [hunk 3]
```diff
@@ -73,7 +73,27 @@ public void DefaultEnvironmentProvider_Prefers_ASPNETCORE_ENVIRONMENT_Over_DOTNE
...
+
+            var value = provider.GetEnvironmentVariable(variableName, "fallback");
+
+            Assert.Equal(string.Empty, value);
+        }
+        finally
+        {
+            Environment.SetEnvironmentVariable(variableName, null);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (akrock) [reply ForgeTrust.AppSurface.Core.Tests/DefaultEnvironmentProviderTests.cs:94]: Addressed in `3291f3c`: added `GetEnvironmentVariable_ReturnsDefault_WhenVariableIsUnset` to cover the unset/null fallback branch. Core tests now pass with 113 tests.

File: ForgeTrust.AppSurface.Core/Defaults/DefaultEnvironmentProvider.cs [hunk 4]
```diff
@@ -43,12 +43,12 @@ public DefaultEnvironmentProvider()
     /// Gets the value of an environment variable from the system.
     /// </summary>
     /// <param name="name">The name of the environment variable.</param>
-    /// <param name="defaultValue">The default value to return if the environment variable is null or empty.</param>
-    /// <returns>The environment variable value or the provided default if the variable is null or empty.</returns>
+    /// <param name="defaultValue">The default value to return if the environment variable is unset.</param>
+    /// <returns>The environment variable value, an empty string when explicitly set empty, or the provided default if the variable is unset.</returns>
     public string? GetEnvironmentVariable(string name, string? defaultValue = null)
```
[the remark this answers is not in the document]
Comment (akrock) [reply ForgeTrust.AppSurface.Core/Defaults/DefaultEnvironmentProvider.cs:56]: Addressed in `3291f3c`: expanded `DefaultEnvironmentProvider.GetEnvironmentVariable` XML docs with null/empty-name semantics, unset vs empty-string behavior, usage guidance, and pitfalls around sensitive values and platform-specific casing.

File: ForgeTrust.AppSurface.Core/ModuleDependencyBuilder.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
+﻿namespace ForgeTrust.AppSurface.Core;
+
+/// <summary>
+/// A builder used to discover and register module dependencies recursively.
+/// </summary>
+public sealed class ModuleDependencyBuilder
```
[the remark this answers is not in the document]
Comment (akrock) [reply ForgeTrust.AppSurface.Core/ModuleDependencyBuilder.cs]: Addressed in `3291f3c`: documented that `ModuleDependencyBuilder` is intended for single-threaded startup composition and is not thread-safe; callers must synchronize concurrent access.

File: ForgeTrust.AppSurface.Core/ModuleDependencyBuilder.cs [hunk 6]
```diff
@@ -1,31 +1,31 @@
...
+    public IEnumerable<IAppSurfaceModule> Modules => _modules.Values;
 
     /// <summary>
     /// Adds a module of type <typeparamref name="T"/> and its dependencies to the builder.
     /// </summary>
     /// <typeparam name="T">The type of the module to add.</typeparam>
     /// <returns>The current <see cref="ModuleDependencyBuilder"/> instance.</returns>
     public ModuleDependencyBuilder AddModule<T>()
-        where T : IRunnableModule, new()
+        where T : IAppSurfaceModule, new()
```
[the remark this answers is not in the document]
Comment (akrock) [reply ForgeTrust.AppSurface.Core/ModuleDependencyBuilder.cs:40]: Addressed in `3291f3c` / prior follow-up commits: `ModuleDependencyBuilder` docs now cover pre-registration, cycle handling, `new()` activation, thread-safety, and the partial-graph behavior if dependency registration throws.

File: Web/ForgeTrust.AppSurface.Docs.Tests/DocMetadataFactoryTests.cs [hunk 7]
```diff
@@ -240,16 +241,16 @@ public void DeriveComponentFromNamespace_ShouldReturnOwningComponent(string name
...
     [Fact]
-    public void DeriveComponentFromNamespace_ShouldReturnRunnableForBareRunnableNamespace()
+    public void DeriveComponentFromNamespace_ShouldReturnAppSurfaceForBareAppSurfaceNamespace()
     {
-        var component = DocMetadataFactory.DeriveComponentFromNamespace("ForgeTrust.Runnable.");
+        var component = DocMetadataFactory.DeriveComponentFromNamespace("ForgeTrust.AppSurface.");
 
-        Assert.Equal("Runnable", component);
+        Assert.Equal("AppSurface", component);
     }
```
[the remark this answers is not in the document]
Comment (akrock) [reply Web/ForgeTrust.AppSurface.Docs.Tests/DocMetadataFactoryTests.cs:257]: Addressed in `3291f3c`: changed the bare namespace test to use exact `ForgeTrust.AppSurface`, and added a separate trailing-dot prefix test for `ForgeTrust.AppSurface.` so both branches are covered.

File: ForgeTrust.AppSurface.Core/ModuleDependencyBuilder.cs [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+        var type = typeof(T);
+        if (!_modules.ContainsKey(type))
+        {
+            var newModule = new T();
+            _modules.Add(type, newModule);
+            newModule.RegisterDependentModules(this);
+        }
+
+        return this;
+    }
```
[the remark this answers is not in the document]
Comment (akrock) [reply ForgeTrust.AppSurface.Core/ModuleDependencyBuilder.cs:51]: Addressed in `3291f3c`: documented the partial-graph sharp edge when `RegisterDependentModules` throws after pre-registration, including the recommendation to treat that builder instance as failed.

File: Dependency/ForgeTrust.AppSurface.Dependency.Autofac.Tests/AppSurfaceAutofacModuleTests.cs [hunk 9]
```diff
@@ -80,9 +80,49 @@ public void RegisterImplementations_RegistersConcreteNonAbstractImplementationsF
...
+                [new FileNotFoundException("missing optional dependency")]));
+
+        using var container = builder.Build();
+
+        var services = container.Resolve<IEnumerable<IScannedAutofacService>>().ToArray();
+
+        Assert.Single(services);
+        Assert.IsType<FirstScannedAutofacService>(services[0]);
+    }
 }
```
[the remark this answers is not in the document]
Comment (akrock) [reply Dependency/ForgeTrust.AppSurface.Dependency.Autofac.Tests/AppSurfaceAutofacModuleTests.cs:145]: Addressed in 0517cf0 by adding RegisterImplementations_ThrowsArgumentNullException_WhenGetTypesIsNull to cover the null getTypes guard.

File: Dependency/ForgeTrust.AppSurface.Dependency.Autofac/AppSurfaceAutofacExtensions.cs [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+    public static IRegistrationBuilder<object, ScanningActivatorData, DynamicRegistrationStyle>
+        RegisterImplementations<TInterface>(this ContainerBuilder builder)
+        where TInterface : notnull
+    {
+        return builder.RegisterImplementations<TInterface>(static assembly => assembly.GetTypes());
+    }
+
+    internal static IRegistrationBuilder<object, ScanningActivatorData, DynamicRegistrationStyle>
+        RegisterImplementations<TInterface>(this ContainerBuilder builder, Func<Assembly, Type[]> getTypes)
+        where TInterface : notnull
```
[the remark this answers is not in the document]
Comment (akrock) [reply Dependency/ForgeTrust.AppSurface.Dependency.Autofac/AppSurfaceAutofacExtensions.cs:60]: Addressed in 0517cf0 by documenting the internal test seam, the non-null getTypes contract, and partial ReflectionTypeLoadException recovery behavior on the overload and helper.

File: ForgeTrust.AppSurface.Core/ModuleDependencyBuilder.cs [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+public sealed class ModuleDependencyBuilder
+{
+    private readonly Dictionary<Type, IAppSurfaceModule> _modules = new();
+
+    /// <summary>
+    /// Gets the collection of registered modules.
+    /// </summary>
+    /// <remarks>
+    /// Modules are keyed by concrete type and added once. Enumeration follows insertion order for the current
+    /// dependency graph, with a module visible in <see cref="Modules"/> before its
```
[the remark this answers is not in the document]
Comment (akrock) [reply ForgeTrust.AppSurface.Core/ModuleDependencyBuilder.cs]: Addressed in 0517cf0 by revising the Modules docs to say enumeration order is implementation-dependent and should not be treated as a stable contract.

File: Dependency/ForgeTrust.AppSurface.Dependency.Autofac/AppSurfaceAutofacExtensions.cs [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+    /// </summary>
+    /// <param name="assembly">The assembly being scanned.</param>
+    /// <param name="getTypes">The delegate used to load the assembly's type array.</param>
+    /// <returns>All loaded types, or the non-null subset from <see cref="ReflectionTypeLoadException.Types"/>.</returns>
+    private static IEnumerable<Type> GetLoadableTypes(Assembly assembly, Func<Assembly, Type[]> getTypes)
+    {
+        try
+        {
+            return getTypes(assembly);
+        }
```
[the remark this answers is not in the document]
Comment (akrock) [reply Dependency/ForgeTrust.AppSurface.Dependency.Autofac/AppSurfaceAutofacExtensions.cs:96]: Addressed in 9d95e47 by adding a deterministic InvalidOperationException when the type loader returns null, documenting that behavior, and adding a regression test for the null return branch.
